File: C/beadando/bead.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#define RESET       "\033[0m" 
#define BG_BLACK    "\033[40m" 
#define BG_RED      "\033[41m" 
#define BG_GREEN    "\033[42m" 
#define BG_YELLOW   "\033[43m" 
#define BG_BLUE     "\033[44m" 
#define BG_MAGENTA  "\033[45m" 
#define BG_CYAN     "\033[46m" 
#define BG_WHITE    "\033[47m"

#define meret 30

#define TERMINAL_CLEAR  "\033[2J" 
#define TERMINAL_HOME   "\033[2H"

#define GIF_FRAME_COUNT 10

#pragma region Szinek a konzolon
enum color {BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE};

typedef enum color color;

void color_print(color c) {
    switch (c) {
        case BLACK:
            printf("%s ", BG_BLACK);
            break;
        case RED:
            printf("%s ", BG_RED);
            break;
        case GREEN:
            printf("%s ", BG_GREEN);
            break;
        case YELLOW:
            printf("%s ", BG_YELLOW);
            break;
        case BLUE:
            printf("%s ", BG_BLUE);
            break;
        case MAGENTA:
            printf("%s ", BG_MAGENTA);
            break;
        case CYAN:
            printf("%s ", BG_CYAN);
            break;
        case WHITE:
            printf("%s ", BG_WHITE);
            break;
    }
    printf("%s", RESET);
}
#pragma endregion

#pragma region Kepek fajlbol
struct image {color** matrix; int width; int height;};

typedef struct image image;

image read_image(char *filename)
{
    FILE *file = fopen(filename, "r");
    image kep;
    fscanf(file, "%d %d", &kep.width, &kep.height);

    if (kep.width > meret || kep.height > meret)
    {
        printf("Tul nagy a meret!");
        fclose(file);
        exit(1);
    }

    kep.matrix = (color**)malloc(kep.height * sizeof(color*));
    for (int i = 0;i<kep.height;++i)
    {
        kep.matrix[i] = (color *)malloc(kep.width * sizeof(color));
    }

    for (int i = 0;i<kep.height;++i)
    {
        for (int j = 0;j<kep.width;++j)
        {
            int c;
            fscanf(file, "%d", &c);
            kep.matrix[i][j] = /* (color) */c;
        }
    }

    fclose(file);
    return kep;
}

void image_print(image img)
{
    for (int i = 0; i<img.height;++i)
    {
        for (int j = 0;j<img.width;++j)
        {
            color_print(img.matrix[i][j]);
        }
        printf("\n");
    }
}

void free_img(image img)
{
    for (int i = 0;i<img.height;++i){
        free(img.matrix[i]);
    }
    free(img.matrix);
}
#pragma endregion

#pragma region Gif fajlbol
typedef struct {image frames[GIF_FRAME_COUNT];} gif;

gif load_gif(char* base_filename) {
    gif g;
    for (int i = 0; i < GIF_FRAME_COUNT; ++i) {
        char file_name[50];
        snprintf(file_name, sizeof(file_name), "%s.bg%d", base_filename, i);

        g.frames[i] = read_image(file_name);
    }
    return g;
}

void print_gif(gif g) {
    for (int i = 0; i < GIF_FRAME_COUNT; ++i) {
        printf(TERMINAL_CLEAR);
        printf(TERMINAL_HOME);
        image_print(g.frames[i]);
        usleep(200000);
    }
}

void free_gif(gif g) {
    for (int i = 0; i < GIF_FRAME_COUNT; ++i) {
        free_img(g.frames[i]);
    }
}
#pragma endregion

int main(void)
{
    color_print(YELLOW);
    printf("\n");


    image kep = read_image("input.txt");
    image_print(kep);
    free_img(kep);


    char base_filename[50];
    printf("Add meg a fajl nevet (pl. 'input'): ");
    scanf("%s", base_filename);

    gif g = load_gif(base_filename);
    print_gif(g); 
    free_gif(g); 

    return 0;
}

File: projektek/C/1/bead.py
import time
from enum import Enum

MAX_SIZE = 30

#region Szinek a konzolon
class Color(Enum):
    RESET = "\033[0m"
    BLACK = "\033[40m"
    RED = "\033[41m"
    GREEN = "\033[42m"
    YELLOW = "\033[43m"
    BLUE = "\033[44m"
    MAGENTA = "\033[45m"
    CYAN = "\033[46m"
    WHITE = "\033[47m"

def color_print(color):
    print(f"{color.value} ", end=Color.RESET.value)

#endregion

#region Kepek fajlbol
class Image:
    def __init__(self, width, height, matrix):
        self.width = width
        self.height = height
        self.matrix = matrix

def image_print(img):
    for item in img.matrix:
        for color in item:
            color_print(color)
        print()

def read_image(file_path):
    with open(file_path, 'r') as f:
        width = int(f.readline())
        height = int(f.readline())
        
        if width > MAX_SIZE or height > MAX_SIZE:
            print("Tul nagy a kep!")
        
        matrix = []
        for _ in range(height):
            row = list(map(int, f.readline().split()))
            matrix.append([
                Color.BLACK if x == 0 else
                Color.RED if x == 1 else
                Color.GREEN if x == 2 else
                Color.YELLOW if x == 3 else
                Color.BLUE if x == 4 else
                Color.MAGENTA if x == 5 else
                Color.CYAN if x == 6 else
                Color.WHITE
                for x in row
            ])
    return Image(width, height, matrix)

#endregion

#region Gif fajlbol
class Gif:
    def __init__(self, frames):
        self.frames = frames

def print_gif(gif):
    for frame in gif.frames:
        print("\033[2J\033[H", end="")
        image_print(frame)
        time.sleep(0.2)

def load_gif(file):
    frames=[]
    for i in range(10):
        file_name = f"{file}.bg{i}"
        frames.append(read_image(file_name))
    return Gif(frames)

#endregion

#region main
color_print(Color.YELLOW)
print()

image = read_image("input.txt")
image_print(image)

base = input("Add meg a fajl nevet (pl. 'input'): ")

gif = load_gif(base)
print_gif(gif)

#endregion
